jena-fuseki2/jena-fuseki-ui/src/services/fuseki.service.js

```diff
@@ -27,19 +27,40 @@ const DATASET_COUNT_GRAPH_QUERY_1 = 'select (count(*) as ?count) {?s ?p ?o}'
 const DATASET_COUNT_GRAPH_QUERY_2 = 'select ?g (count(*) as ?count) {graph ?g {?s ?p ?o}} group by ?g'
 
 class FusekiService {
-  constructor () {
+  /**
+   * @param {Location} location
+   */
+  constructor (location) {
     this.isOffline = true
+    this.pathname = location.pathname
+  }
+
+  /**
+   * Gets the Fuseki URL, only the pathname onward. The protocol, server, port, etc,
+   * are left to the browser/JS engine & Vue to choose. Previously we were passing
+   * strings such as `/#/ping$`. But this did not work when the application was
+   * deployed on a Tomcat server, for example, where the base URL could be something
+   * like `http://localhost:8080/jena-fuseki/#/`.
+   *
+   * See RC1 vote thread for 4.4.0 for more: https://lists.apache.org/thread/z3gb5w95oc7c4v0g1jpk9jkxm0l4b7lh
+   *
+   * @param {string} url - Vue route URL
+   * @return {string} a new route URL that includes any pathname in the URL
+   */
+  getFusekiUrl (url) {
+    // modified version from: https://stackoverflow.com/a/24381515
+    return `${this.pathname}/${url}`.replace(/(\/)\/+/g, '$1')
```

Comment:
We are not using the complete URL, i.e. no chance to have `http://` in the values, as we are only dealing with the `pathname` and the query string passed by Vue.

You can open some URL's in the browser like `localhost`, and `localhost/path`, and `localhost/path/#/` etc, to try, just type `window.location.pathname` in the browser console to see the value.